### ISSUE TITLE
False positive no-member with io and other standard Python modules on Python 3.11

### ISSUE THREAD

**Issue by andy-maier** [PyCQA/pylint#7972]
### Bug description

Pylint below 2.15 on Python 3.11 reports what I consider false positives for the standard Python `io`and `os` packages. The examples below demonstrate this for `io.open()`, but it shows up for multiple members of the `io` and `os` packages.

Test file `io_open.py`:

```
"""
Pylint issue: False positive "no-member" on importing io.open
"""

import io

with io.open('./io_open.py', 'r', encoding='utf-8') as f:
    print(f.read())
```

The error that is reported, is:

```
************* Module io_open
io_open.py:7:5: E1101: Module 'io' has no 'open' member (no-member)
```

The no-member error is reported only for some Python versions and for some pylint versions. From the table below, it seems it is tied to Python 3.11 and pylint below version 2.15.0. The "issue" column of the table indicates whether the no-member error is reported. These tests were all run on macOS 11.7 with Python versions from Homebrew:

```
Python      pylint     astroid    issue
-----------------------------------------

3.10.9      2.10.0     2.7.3      No
3.10.9      2.13.0     2.11.7     No
3.10.9      2.14.0     2.11.7     No
3.10.9      2.15.0     2.12.13    No

3.11.0      2.13.0     2.11.7     Yes
3.11.0      2.14.0     2.11.7     Yes
3.11.0      2.14.5     2.11.7     Yes
3.11.0      2.15.0     2.12.13    No
3.11.0      2.15.9     2.12.13    No
```

Note: The tests on Python 3.11 cannot go to older pylint versions because pylint <2.13 does not install on Python 3.11 due to pinning wrapt to <1.14 (wrapt 1.14.0 has the fix for Python 3.11 removing `inspect.formatargspec`).

From the change log of pylint 2.15.x I could not figure out what would have caused the error to no longer be reported, so I think this behavior should be followed up on.

I first thought this might be related to my other no-member issue #7919, but when looking at `io.py` in Python 3.11, it was not using wildcard imports, in fact it imported `open` explicitly from `_io`.

### Configuration

```ini
# no .pylintrc file, no options used
```

### Command used

```shell
pylint io_open.py
```

### Pylint output

```shell
************* Module io_open
io_open.py:7:5: E1101: Module 'io' has no 'open' member (no-member)
```

### Expected behavior

no-member should not be reported

### Pylint version

```shell
See table in description
```

### OS / Environment

macOS 11.7

### Additional dependencies

For Python 3.11.0 when having installed pylint 2.14.0 (https://github.com/pywbem/pywbem project)

alabaster==0.7.9
appdirs==1.4.3
appnope==0.1.0
astroid==2.11.7
attrs==19.2.0
autodocsumm==0.2.5
Babel==2.7.0
backcall==0.2.0
bleach==3.3.0
build==0.5.1
certifi==2019.11.28
cffi==1.14.5
chardet==3.0.4
click==8.0.2
colorama==0.4.0
coverage==5.0
coveralls==2.1.2
cryptography==3.4.7
Cython==0.29.22
decorator==4.0.11
dill==0.3.6
distlib==0.3.4
docopt==0.6.1
docutils==0.14
dparse==0.6.2
easy-server==0.8.0
easy-vault==0.7.0
entrypoints==0.3
filelock==3.0.0
flake8==4.0.0
FormEncode==2.0.0
future==0.18.2
gitdb2==2.0.0
GitPython==2.1.1
idna==2.5
imagesize==0.7.1
importlib-metadata==1.1.0
iniconfig==1.1.1
ipykernel==4.5.2
ipython==7.16.3
ipython-genutils==0.1.0
ipywidgets==5.2.2
isort==4.3.8
jedi==0.17.2
Jinja2==2.10.2
jsonschema==2.6.0
jupyter==1.0.0
jupyter-client==4.4.0
jupyter-console==6.0.0
jupyter-core==4.2.1
keyring==18.0.0
lazy-object-proxy==1.4.3
lxml==4.9.2
MarkupSafe==1.1.0
mccabe==0.6.0
mistune==0.8.1
more-itertools==4.0.0
nbconvert==5.0.0
nbformat==4.2.0
nocasedict==1.0.1
nocaselist==1.0.3
notebook==4.3.1
packaging==21.0
pandocfilters==1.4.1
parso==0.7.0
pep517==0.13.0
pexpect==4.2.1
pickleshare==0.7.4
pip-check-reqs==2.4.0
pipdeptree==2.2.0
pkginfo==1.4.2
platformdirs==2.2.0
pluggy==0.13.0
ply==3.10
prompt-toolkit==2.0.1
psutil==5.9.0
ptyprocess==0.5.1
py==1.11.0
pycodestyle==2.8.0
pycparser==2.21
pyflakes==2.4.0
Pygments==2.7.4
pyinstrument==3.0.1
pyinstrument-cext==0.2.2
pylint==2.14.0
pyparsing==3.0.7
pyrsistent==0.14.0
pytest==6.2.5
pytest-cov==2.7.0
pytest-easy-server==0.8.0
pytz==2019.1
-e git+ssh://git@github.com/pywbem/pywbem.git@aeadc6bca13389c354edcc938651b7c31dee22a7#egg=pywbem
PyYAML==5.3.1
pyzmq==23.2.1
qtconsole==4.2.1
readme-renderer==23.0
requests==2.25.0
requests-mock==1.6.0
requests-toolbelt==0.8.0
ruamel.yaml==0.17.21
safety==2.2.0
six==1.16.0
smmap2==2.0.1
snowballstemmer==1.2.1
Sphinx==4.2.0
sphinx-git==10.1.1
sphinx-rtd-theme==0.5.0
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-fulltoc==1.2.0
sphinxcontrib-htmlhelp==2.0.0
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.5
sphinxcontrib-websupport==1.1.2
tabulate==0.8.3
terminado==0.6
testfixtures==6.9.0
testpath==0.3
toml==0.10.0
tomlkit==0.10.1
tornado==4.4.2
tox==2.5.0
tqdm==4.14.0
traitlets==4.3.1
twine==3.0.0
urllib3==1.26.5
virtualenv==20.0.35
wcwidth==0.1.7
webencodings==0.5.1
widgetsnbextension==1.2.6
wrapt==1.14.0
yagot==0.5.0
yamlloader==0.5.5
zipp==1.2.0

**Comment by DanielNoord** [comment]
I don't really understand the issue here? On the recent versions this has been fixed right?
This just looks like old versions of pylint aren't compatible with 3.11, which is something we already knew and was never advertised.

**Comment by Pierre-Sassoulas** [comment]
Right, we are not supporting python 3.11 on pylint < 2.15  (and we're not going to). I'm going to close as won't fix

**Comment by andy-maier** [comment]
I agree with not supporting Python 3.11 for pylint <2.15.

The reason I opened the issue was that I could not find anything in the change log of pylint 3.15 that specifically described a solution to this issue, so I thought it might have been fixed by accident, and the fix might be related to issue #7919. Which does not seem to be the case, given that issue #7919 shows up on all Python versions.

Nothing more to be done on this issue here, and apologies for the extra work.

**Comment by Lucas-C** [comment]
I have the same issue with Python 3.11 & Pylint v3.0.0-a5, when using it as a pre-commit hook:
https://github.com/Lucas-C/pre-commit-hooks-safety/actions/runs/3836258509/jobs/6530235042
```
[INFO] Installing environment for https://github.com/pycqa/pylint.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/Lucas-C/pre-commit-hooks-safety.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
pylint...................................................................Failed
- hook id: pylint
- exit code: 2

************* Module pre_commit_hooks.safety_check
pre_commit_hooks/safety_check.py:76:26: E1101: Module 'os' has no 'path' member (no-member)
pre_commit_hooks/safety_check.py:76:39: E1101: Module 'os' has no 'environ' member (no-member)
pre_commit_hooks/safety_check.py:77:15: E1101: Module 'os' has no 'path' member (no-member)
pre_commit_hooks/safety_check.py:78:30: E1101: Module 'os' has no 'path' member (no-member)
pre_commit_hooks/safety_check.py:78:43: E1101: Module 'os' has no 'environ' member (no-member)
pre_commit_hooks/safety_check.py:88:8: E1101: Module 'os' has no 'remove' member (no-member)
pre_commit_hooks/safety_check.py:97:15: E1101: Module 'os' has no 'getcwd' member (no-member)
pre_commit_hooks/safety_check.py:98:4: E1101: Module 'os' has no 'chdir' member (no-member)
pre_commit_hooks/safety_check.py:102:8: E1101: Module 'os' has no 'chdir' member (no-member)

-----------------------------------
Your code has been rated at 6.48/10
```

**Comment by DanielNoord** [comment]
`pylint` `3.0.0` should not be used on `3.11`. In fact the `2.x` are more up to date currently than the beta pylint version.
